github/github.go

```diff
@@ -542,6 +569,25 @@ func (r RateLimits) String() string {
 	return Stringify(r)
 }
 
+type rateLimitCategory uint8
+
+const (
+	coreCategory rateLimitCategory = iota
+	searchCategory
+
+	categories // An array of this length will be able to contain all rate limit categories.
+)
+
+// category returns the rate limit category of the endpoint, determined by Request.URL.Path.
+func category(path string) rateLimitCategory {
+	switch {
+	default:
```

Comment:
Curious... why do you list `default:` first? I don't remember seeing it that way before.